kv/kvserver/concurrency: TestConcurrencyManagerBasic failed

kv/kvserver/concurrency.TestConcurrencyManagerBasic [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/21096169?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/21096169?buildTab=artifacts#/) on master @ [748b6977fe2a4f17eb9278632405b9a52380c0ae](https://github.com/cockroachdb/cockroach/commits/748b6977fe2a4f17eb9278632405b9a52380c0ae):


```
          sequence req=req33
          ----
          [33] sequence req33: sequencing request
          [33] sequence req33: acquiring latches
          [33] sequence req33: waiting to acquire read latch ‹b›@9223372036.854775807,2147483647 for request Get(Shared,Unreplicated) [‹"b"›], [txn: 00000001], held by write latch ‹b›@10.000000000,1 for request Put [‹"b"›], [txn: 00000001]
          [33] sequence req33: blocked on select in spanlatch.(*Manager).waitForSignal
    concurrency_manager_test.go:106: 
        /home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/1301/execroot/com_github_cockroachdb_cockroach/bazel-out/aarch64-fastbuild/bin/pkg/kv/kvserver/concurrency/concurrency_test_/concurrency_test.runfiles/com_github_cockroachdb_cockroach/pkg/kv/kvserver/concurrency/testdata/concurrency_manager/shared_locks_latches:623:
          new-request name=req34 txn=txn2 ts=11,1
          get key=c str=shared
          ----
    concurrency_manager_test.go:106: 
        /home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/1301/execroot/com_github_cockroachdb_cockroach/bazel-out/aarch64-fastbuild/bin/pkg/kv/kvserver/concurrency/concurrency_test_/concurrency_test.runfiles/com_github_cockroachdb_cockroach/pkg/kv/kvserver/concurrency/testdata/concurrency_manager/shared_locks_latches:627:
          sequence req=req34
          ----
          [34] sequence req34: sequencing request
          [34] sequence req34: acquiring latches
          [34] sequence req34: waiting to acquire read latch ‹c›@9223372036.854775807,2147483647 for request Get(Shared,Unreplicated) [‹"c"›], [txn: 00000002], held by write latch ‹c›@10.000000000,1 for request Put [‹"c"›], [txn: 00000001]
          [34] sequence req34: blocked on select in spanlatch.(*Manager).waitForSignal
    concurrency_manager_test.go:106: 
        /home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/1301/execroot/com_github_cockroachdb_cockroach/bazel-out/aarch64-fastbuild/bin/pkg/kv/kvserver/concurrency/concurrency_test_/concurrency_test.runfiles/com_github_cockroachdb_cockroach/pkg/kv/kvserver/concurrency/testdata/concurrency_manager/shared_locks_latches:634:
          debug-latch-manager
          ----
          write count: 3
           read count: 3
    concurrency_manager_test.go:106: 
        /home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/1301/execroot/com_github_cockroachdb_cockroach/bazel-out/aarch64-fastbuild/bin/pkg/kv/kvserver/concurrency/concurrency_test_/concurrency_test.runfiles/com_github_cockroachdb_cockroach/pkg/kv/kvserver/concurrency/testdata/concurrency_manager/shared_locks_latches:639:
          debug-advance-clock ts=2123
          ----
    concurrency_manager_test.go:106: 
        /home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/1301/execroot/com_github_cockroachdb_cockroach/bazel-out/aarch64-fastbuild/bin/pkg/kv/kvserver/concurrency/concurrency_test_/concurrency_test.runfiles/com_github_cockroachdb_cockroach/pkg/kv/kvserver/concurrency/testdata/concurrency_manager/shared_locks_latches:642:
          finish req=req29
        
        expected:
          [-] finish req29: finishing request
          [32] sequence req32: latch conflict with 00000001-0000-0000-0000-000000000000 on ‹"a"› for 2123.000s
          [32] sequence req32: scanning lock table for conflicting locks
          [32] sequence req32: sequencing complete, returned guard
        
        found:
          [-] finish req29: finishing request
          [32] sequence req32: have been waiting 5s to acquire read latch ‹a›@9223372036.854775807,2147483647 for request Get(Shared,Unreplicated) [‹"a"›], [txn: 00000003], held by write latch ‹a›@10.000000000,1 for request Put [‹"a"›], [txn: 00000001]
          [32] sequence req32: latch conflict with 00000001-0000-0000-0000-000000000000 on ‹"a"› for 2123.000s
          [32] sequence req32: scanning lock table for conflicting locks
          [32] sequence req32: sequencing complete, returned guard
          [33] sequence req33: have been waiting 5s to acquire read latch ‹b›@9223372036.854775807,2147483647 for request Get(Shared,Unreplicated) [‹"b"›], [txn: 00000001], held by write latch ‹b›@10.000000000,1 for request Put [‹"b"›], [txn: 00000001]
          [33] sequence req33: blocked on select in spanlatch.(*Manager).waitForSignal
          [34] sequence req34: have been waiting 5s to acquire read latch ‹c›@9223372036.854775807,2147483647 for request Get(Shared,Unreplicated) [‹"c"›], [txn: 00000002], held by write latch ‹c›@10.000000000,1 for request Put [‹"c"›], [txn: 00000001]
          [34] sequence req34: blocked on select in spanlatch.(*Manager).waitForSignal
--- FAIL: TestConcurrencyManagerBasic/shared_locks_latches (9.74s)
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
/cc @cockroachdb/kv-triage
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestConcurrencyManagerBasic.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)

</sub>


Jira issue: CRDB-60191

--- a/pkg/kv/kvserver/concurrency/concurrency_manager_test.go
+++ b/pkg/kv/kvserver/concurrency/concurrency_manager_test.go
@@ -699,9 +699,11 @@ func newCluster() *cluster {
 }
 
 func newClusterWithSettings(st *clustersettings.Settings) *cluster {
-	// Set the latch manager's long latch threshold to infinity to disable
-	// logging, which could cause a test to erroneously fail.
+	// Set the latch manager's long latch hold and slow latch request thresholds
+	// to infinity to disable logging, which could cause a test to erroneously
+	// fail when wall-clock time exceeds the threshold.
 	spanlatch.LongLatchHoldThreshold.Override(context.Background(), &st.SV, math.MaxInt64)
+	spanlatch.SlowLatchRequestThreshold.Override(context.Background(), &st.SV, math.MaxInt64)
 	concurrency.UnreplicatedLockReliabilitySplit.Override(context.Background(), &st.SV, true)
 	concurrency.UnreplicatedLockReliabilityMerge.Override(context.Background(), &st.SV, true)
 	concurrency.UnreplicatedLockReliabilityLeaseTransfer.Override(context.Background(), &st.SV, true)

